rust/drivers/datafusion/src/lib.rs

```diff
@@ -223,23 +240,86 @@ impl Optionable for DataFusionConnection {
         _key: Self::Option,
         _value: adbc_core::options::OptionValue,
     ) -> adbc_core::error::Result<()> {
-        todo!()
+        match _key.as_ref() {
+            constants::ADBC_CONNECTION_OPTION_CURRENT_CATALOG => match _value {
```

Comment:
You can remove the `_` prefix from these args now (https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNUSED_VARIABLES.html#explanation).
```suggestion
        match key.as_ref() {
            constants::ADBC_CONNECTION_OPTION_CURRENT_CATALOG => match value {
```